### PR TITLE
Gradle upgrade to 8.2 

### PR DESCRIPTION
**Pull request by kranthise**
* Also some third party library upgrades

### CODE REVIEW

File: src/main/java/org/gosulang/gradle/GosuPlugin.java [hunk 1]
```diff
@@ -29,26 +29,26 @@ public void apply(Project project) {
...
-    SourceSet test = pluginConvention.getSourceSets().getByName(SourceSet.TEST_SOURCE_SET_NAME);
+    SourceSet main = pluginExtension.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
+    SourceSet test = pluginExtension.getSourceSets().getByName(SourceSet.TEST_SOURCE_SET_NAME);
 
     test.setRuntimeClasspath(project.files(
         test.getOutput(),
         main.getOutput(),
-        project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_CONFIGURATION_NAME),
-        gosuRuntime.inferGosuClasspath(project.getConfigurations().getByName(JavaPlugin.TEST_COMPILE_CONFIGURATION_NAME))));
+        project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_ONLY_CONFIGURATION_NAME),
```
Comment (AlexKorostov) [inline src/main/java/org/gosulang/gradle/GosuPlugin.java:41]: Consider using `testRuntimeClasspath` that automatically combines testRuntime and testImplementation,
see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management (scroll down to see the diagrams of configuration dependencies)

Also there might be an error when you try to resolve TEST_RUNTIME_CONFIGURATION_NAME as it is not resolvable by default, while testRuntimeClasspath is resolvable. Example of this behavior https://stackoverflow.com/questions/47910578/not-able-to-copy-configurations-dependencies-after-upgrading-gradle-plugin-for-a

File: src/main/java/org/gosulang/gradle/tasks/compile/GosuCompile.java [hunk 2]
```diff
@@ -190,7 +190,7 @@ private DefaultGosuCompileSpec createSpec() {
     if (_orderClasspath == null) {
       spec.setClasspath(asList(getClasspath()));
     } else {
-      spec.setClasspath(asList(_orderClasspath.call(project, project.getConfigurations().getByName(JavaPlugin.COMPILE_CONFIGURATION_NAME))));
+      spec.setClasspath(asList(_orderClasspath.call(project, project.getConfigurations().getByName(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME))));
```
Comment (AlexKorostov) [inline src/main/java/org/gosulang/gradle/tasks/compile/GosuCompile.java:193]: Consider using `compileClasspath` configuration instead to support "implementation" (and also "api" if java-library plugin is applied to the project), see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management

File: src/main/java/org/gosulang/gradle/GosuPlugin.java (hunk 1 above)
Comment (kranthise) [reply src/main/java/org/gosulang/gradle/GosuPlugin.java]: Updated the configuration. Thank you

**Comment by DPUkyle** [comment]
This seems out of date; closing as we are using Gradle 8.6 already.

**Pull request closed without merging**